Update extconf

andrewblim opened the pull request:
Mild cleanup of extconf.rb and README/LICENSE. 

Update README to encourage that you install with the `--with-R-dir` flag. This obviates the need to specify R_HOME whenever you want to be able to run RSRuby.

Commit:
Some updates to extconf

- Clean up the handling of --with-R-dir
- Allow --with-R-lib and --with-R-include overrides

Commit:
Remove temporary code

Commit:
Markdownify the readme

Commit:
Move license information to LICENSE.txt

oggy commented on ext/extconf.rb:
```diff
@@ -1,28 +1,49 @@
+r_paths = if $configure_args.has_key?('--with-R-dir')
+  [$configure_args['--with-R-dir']]
+else
+  r_paths = ENV['PATH'].split(File::PATH_SEPARATOR) + DEFAULT_R_PATHS
```
Does this work? I'd expect `PATH` to include `.../bin` dirs, so we couldn't just add `lib` to these.

I think this might work better for the default case (no dir options specified):

```
append_cppflags(`R CMD config --cppflags`.shellsplit)
append_ldflags(`R CMD config --ldflags`.shellsplit)
```

andrewblim replied:
Yeah, that's a much better approach, I'll make that change. Just a heads up that it's a functionality change, if you look at what was there before.

andrewblim replied:
Made some changes, this builds and runs fine for me locally (assuming R_HOME is  _not_ set as an environment variable). If you don't specify flags at build time, we now just expect `R` to be there and for `R CMD` and `R RHOME` to work, and we use that to specify flags as you suggested. RSRUBY_R_HOME no longer defaults to NULL, we expect that `R RHOME` return something.

Commit:
Point to README.md instead of README.txt

Commit:
Rely on `R RHOME` and `R CMD` for Makefile settings

If we don't override with flags like --with-R-lib:

- Use `R RHOME` to set RSRUBY_R_HOME
- Use `R CMD` to set library/include flags, using append_cppflags and
  append_ldflags

Also rename config.h to r_config.h, to make its filename consistent with
the R_CONFIG_H definition.

andrewblim commented:
Ping!
